server/ciphersuites.go

```diff
@@ -40,6 +40,9 @@ var cipherMap = map[string]uint16{
 	"TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384": tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
 	"TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305":    tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,
 	"TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305":  tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,
+	"TLS_AES_128_GCM_SHA256":                  tls.TLS_AES_128_GCM_SHA256,
+	"TLS_AES_256_GCM_SHA384":                  tls.TLS_AES_256_GCM_SHA384,
+	"TLS_CHACHA20_POLY1305_SHA256":            tls.TLS_CHACHA20_POLY1305_SHA256,
```

Comment:
since go 1.14 they add `tls.CipherSuites` and a few other nice things, we might find we could use those once we are ready to move to 1.14?

Reply:
Should we looks at adding them higher up? Is this list a preference list and how does it effect 1.3?

Reply:
This is not a list of preference, this is just a map to get the name from cipher number and vice-versa.
The one that has the list of default ciphers is:
```
func defaultCipherSuites() []uint16 {
..
```
which is used to set `tls.Config.CipherSuites`. However, this field comment says:
```
// CipherSuites is a list of supported cipher suites for TLS versions up to
// TLS 1.2. If CipherSuites is nil, a default list of secure cipher suites
// is used, with a preference order based on hardware performance. The
// default cipher suites might change over Go versions. Note that TLS 1.3
// ciphersuites are not configurable.
CipherSuites []uint16
```
Note that with Go 1.13, when I start the server and connect a go client, I get:
```
[27258] 2020/03/19 09:30:11.559656 [DBG] 127.0.0.1:55645 - cid:1 - TLS version 1.3, cipher suite TLS_AES_128_GCM_SHA256
```
which as you can see is one if the 1.3 cipher that is not in our "defaultCiphers" list.

Reply:
.. which means that we should probably not set our own default ciphers and let go pick them from us, which it does also based on hardware...

Reply:
ok thanks for checking.